src/NLog/NLogTraceListener.cs

```diff
@@ -251,20 +248,32 @@ public override void TraceData(TraceEventCache eventCache, string source, TraceE
             if (Filter != null && !Filter.ShouldTrace(eventCache, source, eventType, id, string.Empty, null, null, data))
                 return;
 
-            var sb = new StringBuilder();
-            for (int i = 0; i < data.Length; ++i)
+            string message = string.Empty;
+            if (data?.Length > 0)
             {
-                if (i > 0)
+                if (data.Length == 1)
                 {
-                    sb.Append(", ");
+                    message = "{0}";
+                }
+                else
+                {
+                    var sb = new StringBuilder();
+                    for (int i = 0; i < data.Length; ++i)
+                    {
+                        if (i > 0)
+                        {
+                            sb.Append(", ");
+                        }
+
+                        sb.Append("{");
+                        sb.Append(i);
+                        sb.Append("}");
+                    }
+                    message = sb.ToString();
```

Comment:
also, isn't string.join more efficient in this way?

Reply:
Not sure I understand string.Join, but I have now changed to use StringBuilderExt.AppendInvariant

Reply:
I was wrong, but for the reference: https://stackoverflow.com/questions/585860/string-join-vs-stringbuilder-which-is-faster#585897